internal/service/projectapikey/resource_project_api_key.go

```diff
@@ -126,15 +128,14 @@ func resourceMongoDBAtlasProjectAPIKeyCreate(ctx context.Context, d *schema.Reso
 	}
 
 	d.SetId(conversion.EncodeStateID(map[string]string{
-		"api_key_id": apiKey.ID,
+		"api_key_id": apiKey.GetId(),
```

Comment:
Looks like getters are missing on the lines above:
`apiKey.GetPublicKey()` and `apiKey.GetPrivateKey()`

Reply:
Done